modules/Kconfig.cmsis

```diff
@@ -22,7 +22,7 @@ endif
 
 menuconfig CMSIS_DSP
 	bool "CMSIS-DSP Library Support"
-	depends on (CPU_CORTEX && NEWLIB_LIBC) || ARCH_POSIX
+	depends on NEWLIB_LIBC || ARCH_POSIX
```

Comment:
Not a blocker, but a note here is that depends on NEWLIBC should be revisited when PICOLIBC becomes the thing